File: Fase 2/revistas-app/src/app/objetos/editor/NumeroRevista.ts
export class NumeroRevista {
    private _numero: number|null;
    private _idRevista: number;
    private _nombre: string;
    private _descripcion: string|null;
    private _fechaPublicacion: Date;
    private _PDF: string;
    private _costoHosting: number;
    private _restriccionMeGusta: string|null;
    private _restriccionComentarios: string|null;
    
    constructor(numero: number|null, idRevista: number, nombre: string, 
                descripcion: string|null, fechaPublicacion: Date, PDF: string, 
                costoHosting: number, restriccionMeGusta: string|null, restriccionComentarios: string|null) {

        this._numero = numero;
        this._idRevista = idRevista;
        this._nombre = nombre;
        this._descripcion = descripcion;
        this._fechaPublicacion = fechaPublicacion;
        this._PDF = PDF;
        this._costoHosting = costoHosting;
        this._restriccionMeGusta = restriccionMeGusta;
        this._restriccionComentarios = restriccionComentarios;
    }
    
    public get numero(): number|null {
        return this._numero;
    }
    public get idRevista(): number {
        return this._idRevista;
    }
    public get nombre(): string {
        return this._nombre;
    }
    public get descripcion(): string|null {
        return this._descripcion;
    }
    public get fechaPublicacion(): Date {
        return this._fechaPublicacion;
    }
    public get PDF(): string {
        return this._PDF;
    }
    public get costoHosting(): number {
        return this._costoHosting;
    }
    public get restriccionMeGusta(): string | null {
        return this._restriccionMeGusta;
    }
    public set restriccionMeGusta(value: string|null) {
        this._restriccionMeGusta = value;
    }
    public get restriccionComentarios(): string | null {
        return this._restriccionComentarios;
    }
    public set restriccionComentarios(value: string|null) {
        this._restriccionComentarios = value;
    }
}

File: Fase 2/ServidorRevistas/src/main/java/objetos/NumeroRevista.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package objetos;

import java.util.Date;

/**
 *
 * @author fernanrod
 */
public class NumeroRevista {
    private Integer numero;
    private Integer idRevista;
    private String nombre;
    private String descripcion;
    private Date fechaPublicacion;
    private String PDF;
    private Float costoHosting;
    private String restriccionMeGusta;
    private String restriccionComentarios;
    
    public NumeroRevista(Integer numero, Integer idRevista, String nombre, String descripcion, 
                        Date fechaPublicacion, String PDF, Float costoHosting, String restriccionMeGusta, String restriccionComentarios) {
        this.numero = numero;
        this.idRevista = idRevista;
        this.nombre = nombre;
        this.descripcion = descripcion;
        this.fechaPublicacion = fechaPublicacion;
        this.PDF = PDF;
        this.costoHosting = costoHosting;
        this.restriccionMeGusta = restriccionMeGusta;
        this.restriccionComentarios = restriccionComentarios;
    }
    
    public Integer getNumero() {
        return numero;
    }
    
    public Integer getIdRevista() {
        return idRevista;
    }

    public String getNombre() {
        return nombre;
    }

    public String getDescripcion() {
        return descripcion;
    }

    public Date getFechaPublicacion() {
        return fechaPublicacion;
    }

    public String getPDF() {
        return PDF;
    }

    public Float getCostoHosting() {
        return costoHosting;
    }
    
    public String getRestriccionMeGusta() {
        return restriccionMeGusta;
    }

    public String getRestriccionComentarios() {
        return restriccionComentarios;
    }
}
